examples/NSIDC/cmr_serializer.py

```diff
@@ -0,0 +1,98 @@
+from IPython.display import display
+from IPython.core.display import HTML
+
+
+class QueryGranule(object):
+
+    def _filter_related_links(self, filter: str):
+        """
+        Filter RelatedUrls from the UMM fields on CMR
+        """
+        matched_links = []
+        for link in self._data['umm']['RelatedUrls']:
+            if link['Type'] == filter:
+                matched_links.append(link['URL'])
+        return matched_links
+
+    def __repr__(self):
+        data_links = [l for l in self.data_links()]
+        rep_str = f"""
+        Id: {self.id()}
+        Collection: {self._data['umm']['CollectionReference']}
+        Spatial coverage: {self._data['umm']['SpatialExtent']}
+        Temporal coverage: {self._data['umm']['TemporalExtent']}
+        Size(MB): {self.size()}
+        Data: {data_links}\n
+        """
+        return rep_str
+
+    def _repr_html_(self):
+        dataviz_img = ''.join([f'<img src="{l}" width="340px" />' for l in self.dataviz_links()[0:2]])
+        data_links = ''.join([f'<a href="{l}" target="_blank">{l}</a><BR>' for l in self.data_links()])
+        granule_str = f"""
+        <p>
+          <b>Id</b>: {self.id()}<BR>
+          <b>Collection</b>: {self._data['umm']['CollectionReference']}<BR>
+          <b>Spatial coverage</b>: {self._data['umm']['SpatialExtent']}<BR>
+          <b>Temporal coverage</b>: {self._data['umm']['TemporalExtent']}<BR>
+          <b>Size(MB):</b> {self.size()} <BR>
+          <b>Data</b>: {data_links}<BR>
+          <span>{dataviz_img}</span>
+        </p>
+        """
+        return granule_str
```

Comment:
I didn't know that this is how you create custom HTML views that show up in Jupyter. Cool!

Reply:
Yeah we can even create little interactive snippets, the same way xarray does it. I was thinking in packaging these functions into a client-side CMR library and well have even more functionality so users do the same with way way less code, like a NASA Earthdata SDK.